lib/typeprof/core/ast/control.rb

```diff
@@ -339,5 +339,15 @@ def install0(genv)
         ret
       end
     end
+
+    class RetryNode < Node
+      def initialize(raw_node, lenv)
+        super(raw_node, lenv)
+      end
+
+      def install0(genv)
+        Source.new(genv.nil_type)
```

Comment:
It would be better to return `Type::Bot.new(genv)` here instead of `genv.nil_type`. (Bot type means a type that does not return normally.)